src/codegen/codegen_cpp_visitor.cpp

```diff
@@ -319,16 +319,16 @@ void CodegenCVisitor::visit_update_dt(const ast::UpdateDt& node) {
 }
 
 void CodegenCVisitor::visit_protect_statement(const ast::ProtectStatement& node) {
-    printer->fmt_start_block("#pragma omp critical {}", info.mod_suffix);
+    printer->add_line("#pragma omp atomic update");
```

Comment:
note: we will need to update NMODL docs to mention constraints to use `PROTECT` statement (i.e. they need to be valid automic statements for OpenMP)

Reply:
If `CodegenCVisitor` implemented `print_atomic_reduction_pragma` then we wouldn't need to implement `visit_protect_statement` in `CodegenAccVisitor`..right?

Reply:
yes..needed some extra changes done in 98eb8488d3dda85bf1e2fcff7e628c43de1221fd (basically, we don't need to emit `pragma omp atomic` at all places where `print_atomic_reduction_pragma()` is used today).